src/Functions/URL/removeGarbageParametersFromURL.cpp

```diff
@@ -0,0 +1,427 @@
+#include <algorithm>
+#include <vector>
+#include <cmath>
+#include <cstring>
+#include <cstdlib>
+#include <boost/math/distributions/chi_squared.hpp>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsStringUIntToString.h>
+#include <base/find_symbols.h>
+#include "protocol.h"
+
+namespace DB
+{
+
+struct RemoveGarbageParametersFromURLImpl
+{
+    /// @brief Guesses with 95% accuracy whether symbols in string form uniform distribution
+    /// @param c_start Pointer to the first symbol of the string
+    /// @param c_end Pointer to the symbol following the last symbol of the string
+    /// @param a First character of uniform distribution
+    /// @param b Last character of uniform distribution
+    /// @return true - if there is 95%+ probability the string matches uniform distribution with given parameters, false - otherwise
+    static bool pearsonIsUniform(const char * c_start, const char * c_end, char a, char b)
+    {
+        /// Deciding how much intervals to split symbols into and whether the string is long enough to be handled
+        int64_t length = c_end - c_start;
+        int l = static_cast<int>(3.32 * log10(length)) + 1;
```

Comment:
To a random future reader, this looks like too much magic. Could you please document the kind of test/algorithm used in `pearsonIsUniform()?